src/clims/services/extensible_service_api.py

```diff
@@ -31,18 +31,29 @@ def _all_qs(self):
 
         return self._archetype_version_class.objects.filter(latest=True).prefetch_related('properties')
 
-    def filter(self, *args, **kwargs):
-        pass
+    def filter(self, **kwargs):
+        get_args = self._get_filter_arguments(**kwargs)
+        models = self._archetype_version_class.objects.\
+            prefetch_related('properties__extensible_property_type').\
+            filter(**get_args)
+        return [self.to_wrapper(m) for m in models]
 
     def get(self, **kwargs):
-        # TODO I'm not at all sure that this is a perfect method to handle this.
-        #      but for now it might be suff
+        get_args = self._get_filter_arguments(**kwargs)
+        model = self._archetype_version_class.objects.\
+            prefetch_related('properties__extensible_property_type').\
+            get(**get_args)
+        return self.to_wrapper(model)
+
+    def _get_filter_arguments(self, **kwargs):
         get_args = {}
         if 'latest' not in get_args.keys():
             get_args['latest'] = True
         for key, value in kwargs.items():
-            if not key == 'latest':
+            if key == 'project':
+                get_args['archetype__project__name'] = value.name
+            elif key == 'project_name':
+                get_args['archetype__project__name'] = value
+            elif not key == 'latest':
```

Comment:
I believe that we are missing a final `else` clause here.

Reply:
If you look to the original code, the original statement was, if not key == latest. It this was not fullfilled, nothing happened. Handling of 'latest' is a bit weird though, but I decided to wait with that.

Reply:
Yeah. I think that this code is something that we will have to do some additional thinking on, to figure out what is the best approach here. For now I think this does the job. Perhaps it would be easier to understand if we added an `else` clause and added a comment that this is the `latest` case, and that we should just pass that? Or perhaps it is easier to do something like:

```
if key == 'latest:
    pass
elif key == 'project':
 [...]
else:
     get_args['archetype__{}'.format(key)] = value
```